universal-application-tool-0.0.1/app/repository/ProgramRepository.java

```diff
@@ -47,6 +52,50 @@ public Program updateProgramSync(Program program) {
     return program;
   }
 
+  public void updateQuestionVersions(Program program) {
+    Transaction transaction = ebeanServer.beginTransaction();
+    try {
+      updateQuestionVersions(program, transaction);
+      ebeanServer.commitTransaction();
+    } finally {
+      ebeanServer.endTransaction();
+    }
+  }
+
+  public void updateQuestionVersions(Program program, Transaction transaction) {
+    ProgramDefinition.Builder updatedDefinition =
+        program.getProgramDefinition().toBuilder().setBlockDefinitions(ImmutableList.of());
+    for (BlockDefinition block : program.getProgramDefinition().blockDefinitions()) {
+      BlockDefinition.Builder updatedBlock =
+          block.toBuilder().setProgramQuestionDefinitions(ImmutableList.of());
+      for (ProgramQuestionDefinition question : block.programQuestionDefinitions()) {
+        Optional<Question> updatedQuestion =
+            ebeanServer
+                .find(Question.class)
+                .usingTransaction(transaction)
+                .where()
+                .eq(
+                    "name",
```

Comment:
Suggestion (blocking): please ensure questions are indexed on `name`

Reply:
Also we should have a database-level constraint preventing two questions from having the same name and version ID.

Reply:
Thank you. Yep, and who knows if the same code will be running 10 years from now after many version changes.